src/envoy/http/authn/authenticator_base_test.cc

```diff
@@ -82,113 +91,78 @@ Http::TestHeaderMapImpl CreateTestHeaderMap(const std::string& header_key,
 }
 
 TEST_F(AuthenticatorBaseTest, ValidateMtlsOnPlaintextConnection) {
-  iaapi::MutualTls mTlsParams;
-  authenticator_.validateX509(mTlsParams,
-                              [](const Payload* payload, bool success) {
-                                // When requiring mTLS, plaintext connection
-                                // should fail.
-                                EXPECT_FALSE(payload);
-                                EXPECT_FALSE(success);
-                              });
+  EXPECT_FALSE(authenticator_.validateX509(mTlsParams, payload));
```

Comment:
Can you also verify payload is not mutated. Compare with default (empty) payload may be sufficient.